src/mon/MDSMonitor.cc

```diff
@@ -1439,6 +1439,27 @@ bool MDSMonitor::prepare_command(MonOpRequestRef op)
   }
 }
 
+bool MDSMonitor::_has_health_warnings(vector<int> warnings)
+{
+  for (map<uint64_t, MDSHealth>::iterator i = pending_daemon_health.begin();
+       i != pending_daemon_health.end(); ++i) {
+    auto metrics = i->second.metrics;
+    for (vector<MDSHealthMetric>::iterator j = metrics.begin();
+	 j != metrics.end(); ++j) {
+
+      auto health_warn_type = j->type;
+      for (vector<int>::iterator k = warnings.begin(); k != warnings.end();
+	   ++k) {
+	if ((health_warn_type == (*k))) {
+	  return true;
+	}
+      }
```

Comment:
```suggestion
      auto it = std::find(warnings.begin(), warnings.end(), health_warn_type);
      if (it != warnings.end()) return true;
```

Reply:
Ah, right. I forgot about `std::find()`. Thanks! I've incorporated this but slightly differently since according to CodingStyle following style is not acceptable - `if (foo) bar;  // no `.